src/NodeTypeResolver/NodeTypeResolver.php

```diff
@@ -98,6 +101,19 @@ public function isObjectType(Node $node, ObjectType $requiredObjectType): bool
             return false;
         }
 
+        // warn about invalid use of this method
+        if ($node instanceof ClassMethod) {
+            throw new ShouldNotHappenException(
+                'ClassMethod itself does not have any type. Check the Class_/Interface/Trait_ node instead'
+            );
+        }
+
+        if ($node instanceof ClassConst) {
+            throw new ShouldNotHappenException(
+                'Class constant itself does not have any type. Check the Class_/Trait_ instead'
```

Comment:
interface can have constant, trait can have start from php 8.2.0, so the message should similar as above:

```suggestion
                'Class constant itself does not have any type. Check the Class_/Interface/Trait_ node instead'
```

Reply:
:+1:  I'll use "class like" element to keep it simple :)

Reply:
https://github.com/rectorphp/rector-src/pull/6282